src/modules/EventModule.cpp

```diff
@@ -0,0 +1,792 @@
+								embed.setColor(15158332);
+								embed.setTitle(eventEmoji);
+								embed.setDescription("**HighRiskHighReward** event has **ended**!");
+
+								config.isEventRunning = false;
+								config.eventHighRiskHighRewardRunning = false;
+								UmikoBot::Instance().createMessage(chan, embed);
+							}
+							else if (config.eventLowRiskLowRewardRunning)
+							{
+								Embed embed;
+								embed.setColor(15158332);
+								embed.setTitle(eventEmoji);
+								embed.setDescription("**LowRiskLowReward** event has **ended**!");
+
+								config.isEventRunning = false;
+								config.eventLowRiskLowRewardRunning = false;
+								UmikoBot::Instance().createMessage(chan, embed);
+							}
+							if (config.eventRaffleDrawRunning)
+							{
+								Embed embed;
+								embed.setColor(15844367);
+								embed.setTitle(eventEmoji);
+
+								if (config.currentTicketIndex == 0)
+								{
+									embed.setDescription("**RAFFLE DRAW** event has **ended**!\n"
+										"**There was no draw because no one bought tickets**");
+									config.claimedReward = true;
+									config.isEventRunning = false;
+									config.eventRaffleDrawRunning = false;
+									config.eventTimer->stop();
+									UmikoBot::Instance().createMessage(chan, embed);
+									return;
+								}
+
+								//Announce Winner
+								int lastTicket = static_cast<int>(config.currentTicketIndex);
+								std::random_device randomDevice;
+								std::mt19937 gen(randomDevice());
+								std::uniform_int_distribution<> dist(1, lastTicket);
+								int luckyTicketId = dist(gen);
+
+								QString tiketId = QString::number(luckyTicketId);
+								embed.setDescription("**RAFFLE DRAW** event has **ended**!\n"
+									"**The lucky ticket which wins the draw is `" + tiketId + "`**\n"
+									"**The owner can claim their rewards within 24 hours**");
+								config.claimedReward = false;
+								config.currentTicketIndex = 0;
+
+								//Get the user who has the lucky ticket
+								for (auto& user : raffleDrawGuildList[guildID])
+								{
+									for (auto& userTicket : user.m_TicketIds)
+									{
+										if (luckyTicketId == userTicket)
+										{
+											config.luckyUser = user.m_UserId;
+										}
+									}
+								}
+
+								//Clear all the tickets
+								for (auto& user : raffleDrawGuildList[guildID])
+									user.m_TicketIds.clear();
+
+								//Set the reward claiming time (24 hours)
+								config.raffleDrawRewardClaimTimer = new QTimer;
+								config.raffleDrawRewardClaimTimer->setInterval(24 * 3600000);
+								config.raffleDrawRewardClaimTimer->setSingleShot(true);
+
+								QObject::connect(config.raffleDrawRewardClaimTimer, &QTimer::timeout, [this, guildID, chan, authorId]()
+									{
+										auto& config = getServerEventData(guildID);
+										config.claimedReward = true;
+										UmikoBot::Instance().createMessage(chan, "The owner of the price in the RaffleDraw didn't get their rewards!\n"
```

Comment:
Should be "prize" instead of "price".

Reply:
Also use "claim" instead of "get"

Reply:
And it would be better if we don't reference the winner as the owner of the prize because they would have to claim it first to own it.